app/services/doc_auth/mock/yml_loader_concern.rb

```diff
@@ -8,10 +8,12 @@ module YmlLoaderConcern
       def parse_yaml(uploaded_file)
         data = YAML.safe_load(uploaded_file, permitted_classes: [Date])
         if data.is_a?(Hash)
-          if (m = data.dig('document', 'dob').to_s.
-            match(%r{(?<month>\d{1,2})/(?<day>\d{1,2})/(?<year>\d{4})}))
-            data['document']['dob'] =
-              Date.new(m[:year].to_i, m[:month].to_i, m[:day].to_i)
+          ['dob', 'state_id_expiration'].each do |dt|
+            if (m = data.dig('document', dt).to_s.
+              match(%r{(?<month>\d{1,2})/(?<day>\d{1,2})/(?<year>\d{4})}))
+              data['document'][dt] =
+                Date.new(m[:year].to_i, m[:month].to_i, m[:day].to_i)
+            end
           end
```

Comment:
I just realized this is the same regex as `DateParser::AMERICAN_REGEX`, maybe we could reuse that code here?

```suggestion
          %w[dob state_id_expiration].each do |date_key|
            if (date_s = data.dig('document', date_key'))
              data['document'][date_key] = DateParser.parse_legacy(date_s)
            end
          rescue Date::Error
            # maybe don't rescue so we get the error?
          end
```